File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.js
// 문자열 나누기

function solution(s) {
  let answer = 0;
  let targetString = "";
  let sameCount = 0;
  let anotherSameCount = 0;
  for (const item of s) {
    if (!targetString) {
      targetString = item;
      sameCount = 1;
    } else {
      if (targetString !== item) {
        anotherSameCount += 1;
      } else {
        sameCount += 1;
      }

      if (sameCount === anotherSameCount) {
        answer += 1;
        sameCount = 0;
        anotherSameCount = 0;
        targetString = "";
      }
    }
  }
  if (targetString) {
    answer += 1;
  }
  return answer;
}

File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    
    target = s[0]
    targetCount = 0
    diffCount = 0
    
    for i in range(len(s) - 1):
        if target == s[i]:
            targetCount += 1
        else:
            diffCount += 1
        
        if targetCount == diffCount:
            target = s[i + 1]
            answer += 1
            targetCount = 0
            diffCount = 0
    
    return answer + 1